ecl/hql/hqlfold.cpp

```diff
@@ -3525,6 +3525,30 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
                 IHqlExpression * newEqual = createBoolExpr(no_eq, ensureExprType(leftExpr, type), ensureExprType(key, type));
                 return createIf(newEqual, LINK(mapto->queryChild(1)), LINK(expr->queryChild(2)));
             }
+
+            // Special case boolean cases matching constant values
+            if (leftExpr->isBoolean())
+            {
+                HqlExprArray args;
+                bool seenOpt[2] = { false, false };
+                args.append(*LINK(leftExpr));
+                for (unsigned idx = 1; idx <= numCases; idx++)
+                {
+                    IHqlExpression * child = expr->queryChild(idx);
+                    IHqlExpression * grand = child->queryChild(0);
+                    IValue * grandValue = grand->queryValue();
+                    if (grandValue)
+                    {
+                        seenOpt[grandValue->getBoolValue()] = true;
+                        if (seenOpt[false] && seenOpt[true])
```

Comment:
Trying to understand this bit of code...

Reply:
@richardkchapman I think: if it's seen both the true and false case, then it can build the  if expression with the arguments it see so far.

Reply:
That would make sense (but perhaps a comment would have helped)

Reply:
Shamser is correct, I'll add a comment.